src/utils.h

```diff
@@ -35,41 +38,53 @@ extern log_level_t log_level;
 extern char *log_cid;
 extern gboolean use_syslog;
 
+static inline const char *get_errno_string(void)
+{
+	static __thread char errbuf[256];
+	int saved_errno = errno;
+	snprintf(errbuf, sizeof(errbuf), "%s", strerror(saved_errno));
```

Comment:
This does not avoid the usage of `strerror`, so I don’t think this is an improvement compared to the previous version?!

Using `%m` here might work.